modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/SubscriberConnection.java

```diff
@@ -44,6 +44,18 @@
     private String connectedNode;
 
     /**
+     * Used to mark the subscription as obsolete. Upon being set to true, messages that are scheduled to be
+     * delivered will be dropped.
+     */
+    private AtomicBoolean obsolete;
```

Comment:
Since, both methods that are calling this are synchronized, do we need an atomic boolean here?